Controller/Product/Info.php

```diff
@@ -85,26 +85,45 @@ public function execute()
     protected function _initProduct()
     {
         $storeId = $this->_objectManager->get('Magento\Store\Model\StoreManagerInterface')->getStore()->getId();
-        try {
-            $product = false;
+        $product = false;
 
-            if ($sku = $this->getRequest()->getParam('sku')) {
-                $product = $this->productRepository->get($sku, false, $storeId);
-            } elseif ($id = (int)$this->getRequest()->getParam('product', $this->getRequest()->getParam('id', false))) {
-                $product = $this->productRepository->getById($id, false, $storeId);
-            }
+        $id = $sku = $this->getRequest()->getParam('sku', false);
+        if (!$id) {
+            $id = $this->getRequest()->getParam('product', $this->getRequest()->getParam('id', false));
+        }
 
-            if (!$product->getId()) {
-                throw new \Exception('Invalid product.');
-            }
+        $product = $this->getProductBySku($storeId, $id);
```

Comment:
It is quite misleading that this method is called getProductBySku, but then you can pass either the sku or product id.

Reply:
Ah I see on the next line.

I guess I would rather have made it more separate

$sku = getSKU()
if (sku) { $product = getProductBySKU}
else { // fallback
$id = getID;
$product = getProductById()
}

Reply:
I see that for readability its not the best however there was already the functionality to pass in the `sku` and `product` (id) however since we need to support both cases I leave it as it is...